### PR TITLE
Migrate `aws-vpc` to official AWS SDK

### PR DESCRIPTION
**Pull request by kdomanski**
This will discard the unmaintained library [`mitchellh/goamz`](https://www.github.com/mitchellh/goamz) in favor of the [official AWS Go SDK](https://www.github.com/aws/aws-sdk-go).

Currently flannel does not work in the `eu-central-1` AWS region (Frankfurt) because the new region supports only V4-signed API requests. Thus in addition to the obvious maintenance benefits, the commits will fix flannel for Amazon's Frankfurt DC.

### CODE REVIEW

**Comment by eyakubovich** [comment]
LGTM but I'll defer to @MohdAhmad

File: backend/awsvpc/awsvpc.go [hunk 1]
```diff
@@ -209,24 +207,32 @@ func (m *AwsVpcBackend) detectRouteTableID(instanceID string, ec2c *ec2.EC2) err
 	log.Info("Subnet-ID: ", subnetID)
```
Comment (MohdAhmad) [inline backend/awsvpc/awsvpc.go]: *subnetID

File: backend/awsvpc/awsvpc.go [hunk 2]
```diff
@@ -209,24 +207,32 @@ func (m *AwsVpcBackend) detectRouteTableID(instanceID string, ec2c *ec2.EC2) err
...
 	log.Info("VPC-ID: ", vpcID)
```
Comment (MohdAhmad) [inline backend/awsvpc/awsvpc.go]: *vpcID

**Comment by MohdAhmad** [comment]
LGTM apart from the minor changes.

**Comment by kdomanski** [comment]
@eyakubovich @MohdAhmad updated

**Pull request merged**